olive/model/handler/mixin/mlflow.py

```diff
@@ -6,41 +6,62 @@
-        if (Path(self.model_path) / "data" / "model").exists():
-            copy_dir(Path(self.model_path) / "data" / "model", target_path, dirs_exist_ok=True)
-            copy_dir(Path(self.model_path) / "data" / "config", target_path, dirs_exist_ok=True)
-            copy_dir(Path(self.model_path) / "data" / "tokenizer", target_path, dirs_exist_ok=True)
-            return target_path
-        return None
-
-    def get_mlflow_model_path_or_name(self, cache_dir):
-        # both config.json and model file will be saved under data/model
-        mlflow_transformer_model_path = self.to_mlflow_transformer_model(cache_dir)
-        if not mlflow_transformer_model_path:
-            logger.debug(
-                "Model path %s does not exist. Use hf_config.model_name instead.", mlflow_transformer_model_path
-            )
-            return self.hf_config.model_name
-        return str(mlflow_transformer_model_path)
+    def maybe_init_mlflow(self):
+        """Initialize MLFlow model if the model is saved in MLFlow format."""
+        parent_dir = Path(self.get_resource("model_path")).resolve()
+
+        yaml_path = parent_dir / "MLmodel"
+        if not yaml_path.exists():
+            return
+
+        with open(yaml_path) as fp:
+            mlflow_data = yaml.safe_load(fp)
+            # default flavor is "hftransformersv2" from azureml.evaluate.mlflow>=0.0.8
+            # "hftransformers" from azureml.evaluate.mlflow<0.0.8
+            # TODO(trajep): let user specify flavor name if needed
+            # to support other flavors in mlflow not only hftransformers
+            flavors = mlflow_data.get("flavors", {})
+            if not flavors or not any(flavor.startswith("hftransformers") for flavor in flavors):
+                raise ValueError(
+                    "Invalid MLFlow model format. Please make sure the input model"
+                    " format is same with the result of mlflow.transformers.save_model,"
+                    " or aml_mlflow.hftransformers.save_model from azureml.evaluate.mlflow"
+                )
+
+        model_dir = parent_dir / "data" / "model"
+        assert model_dir.is_dir(), "Model directory does not exist."
+
+        config_dir = parent_dir / "data" / "config"
+        tokenizer_dir = parent_dir / "data" / "tokenizer"
+
+        # some mlflow models only have model directory
+        if not config_dir.exists() and not tokenizer_dir.exists():
+            self.mlflow_model_path = str(model_dir)
+            return
+
+        # some mlflow models have config and tokenizer directories but model directory also
+        # contains the same files
+        model_dir_contents = set(model_dir.iterdir())
+        if set(config_dir.iterdir()) <= model_dir_contents and set(tokenizer_dir.iterdir()) <= model_dir_contents:
+            self.mlflow_model_path = str(model_dir)
+            return
+
+        # have to gather all contents into a single directory
+        cache = OliveCache.from_cache_env()
```

Comment:
Seems you removed `mlflow_transformer_model_cache_dir` and set the default mlflow model folder under cache directory to bypass the issue of `read-only` in default local downloading folder we met before, right?

Reply:
Yes. This HfModelHandler automatically prepares mlflow models so the user can just use it like it's a model id or local path. We already have a cache folder and the mlflow model is already downloaded there (when running locally) so it's easier to just copy or hard link there.

Reply:
Just pushed a commit which executes this logic only if we ask for `model_name_or_path`. for most of the common uses cases like loading model, config and tokenizer, we can just call them on the correct subdirectory. 

Otherwise, creating a copy of the model files during init is inefficient if we don't really need it.